haystack/document_stores/types/filter_policy.py

```diff
@@ -264,11 +289,11 @@ def combine_two_comparison_filters(
         runtime_comparison_filter = {"field": "meta.type", "operator": "==", "value": "article"},
         init_comparison_filter = {"field": "meta.date", "operator": ">=", "value": "2015-01-01"},
         new_filters = combine_two_comparison_filters(
-            init_comparison_filter, runtime_comparison_filter, "AND"
+            init_comparison_filter, runtime_comparison_filter, LogicalOperator.AND
         )
         # Output:
         {
-            "operator": "AND",
+            "operator": LogicalOperator.AND,
```

Comment:
Let's keep the string representation in the output dict (to be consistent with the comparison operators)

Reply:
See [519de1c](https://github.com/deepset-ai/haystack/pull/8042/commits/519de1c5edd9f6ab3616ae477ac98d66e2975372) for details

Reply:
Uhm, looks like you only modified the docstring and not the code? The operator enum should be converted to its string repr before being saved to the dict.

Reply:
My mistake, I misunderstood the request, here is the fix [a2621ad](https://github.com/deepset-ai/haystack/pull/8042/commits/a2621ada9c6dd54e170b9268935ae751b9461890)